tests/benchmarks/test_report.py

```diff
@@ -0,0 +1,141 @@
+from shared.reports.readonly import ReadOnlyReport
+from shared.reports.resources import Report
+
+
+def read_fixture(name: str) -> bytes:
+    with open(name, "rb") as f:
+        data = f.read()
+
+    dctx = zstd.ZstdDecompressor()
+    return dctx.decompress(data)
+
+
+READABLE_VARIANTS = [
+    pytest.param(Report, False, id="Report"),
+    pytest.param(ReadOnlyReport, False, id="ReadOnlyReport"),
+    pytest.param(ReadOnlyReport, True, id="Rust ReadOnlyReport"),
+    pytest.param(EditableReport, False, id="EditableReport"),
+]
+
+EDITABLE_VARIANTS = [Report, EditableReport]
+
+
+def init_mocks(mocker, should_load_rust) -> tuple[bytes, bytes]:
+    mocker.patch(
+        "shared.reports.readonly.ReadOnlyReport.should_load_rust_version",
+        return_value=should_load_rust,
+    )
+
+    raw_chunks = read_fixture("tests/benchmarks/fixtures/worker_chunks.txt.zst")
+    raw_report_json = read_fixture("tests/benchmarks/fixtures/worker_report.json.zst")
+
+    return raw_chunks, raw_report_json
+
+
+def do_parse(report_class, raw_report_json, raw_chunks):
+    report_json = orjson.loads(raw_report_json)
+    chunks = raw_chunks.decode()
+    return report_class.from_chunks(
+        chunks=chunks, files=report_json["files"], sessions=report_json["sessions"]
+    )
+
+
+@pytest.mark.parametrize("report_class, should_load_rust", READABLE_VARIANTS)
+def test_report_parsing(report_class, should_load_rust, mocker, benchmark):
+    raw_chunks, raw_report_json = init_mocks(mocker, should_load_rust)
+
+    def bench_fn():
+        do_parse(report_class, raw_report_json, raw_chunks)
+
+    benchmark(bench_fn)
+
+
+@pytest.mark.parametrize("report_class, should_load_rust", READABLE_VARIANTS)
+def test_report_iterate_all(report_class, should_load_rust, mocker, benchmark):
+    raw_chunks, raw_report_json = init_mocks(mocker, should_load_rust)
+
+    report = do_parse(report_class, raw_report_json, raw_chunks)
+
+    def bench_fn():
+        for file in report:
+            for _line in file:
+                pass
+
+    benchmark(bench_fn)
+
+
+@pytest.mark.parametrize("report_class, should_load_rust", READABLE_VARIANTS)
+def test_report_process_totals(report_class, should_load_rust, mocker, benchmark):
+    raw_chunks, raw_report_json = init_mocks(mocker, should_load_rust)
+
+    report = do_parse(report_class, raw_report_json, raw_chunks)
+
+    def bench_fn():
+        report._process_totals()
+        for file in report:
+            file._process_totals()
+
+    benchmark(bench_fn)
+
+
+@pytest.mark.parametrize("report_class, should_load_rust", READABLE_VARIANTS)
+def test_report_filtering(report_class, should_load_rust, mocker, benchmark):
+    raw_chunks, raw_report_json = init_mocks(mocker, should_load_rust)
+
+    report = do_parse(report_class, raw_report_json, raw_chunks)
+
+    def bench_fn():
+        filtered = report.filter(paths=[".*"], flags=["unit"])
+        filtered._process_totals()
```

Comment:
Are these operations necessary for the filtering, or the filter process is a lazy one and you're just forcing the filtered report being parsed?

Reply:
yes, the second one. the `filter` method just creates a wrapper class based on the underlying report, with a different filter on top.
and getting the totals (or iterating the files) is one of the operations that is frequently performed on those reports.

for example, the whole slowness of component comparisons is based on getting the totals / diff totals for the report filtered for the specific component / flag